src/structurizr/api/structurizr_client.py

```diff
@@ -276,13 +276,14 @@ def _add_headers(
 
     def _archive_workspace(self, json: str) -> None:
         """Store the workspace."""
-        location = self._create_archive_filename()
-        logger.debug(
-            f"Archiving workspace {self.workspace_id} to"
-            f" '{self.workspace_archive_location}'."
-        )
-        with gzip.open(location, mode="wt") as handle:
-            handle.write(json)
+        if self.workspace_archive_location is not None:
```

Comment:
I think it's simpler to just exit early:

```suggestion
        if self.workspace_archive_location is None:
            return
```

Reply:
And then adjust indentation level.

Reply:
Yep, fair enough.